pkg/preparer/orchestrate.go

```diff
@@ -150,22 +151,28 @@ func (p *Preparer) handlePods(podChan <-chan pods.Manifest, quit <-chan struct{}
 			if !working {
 				p.tryRunHooks(hooks.AFTER_AUTH_FAIL, pods.NewPod(manifestToLaunch.ID(), pods.PodPath(p.podRoot, manifestToLaunch.ID())), &manifestToLaunch, manifestLogger)
 			}
-		case <-time.After(1 * time.Second):
+		default:
 			if working {
 				pod := pods.NewPod(manifestToLaunch.ID(), pods.PodPath(p.podRoot, manifestToLaunch.ID()))
 
 				// HACK ZONE. When we have better authz, rewrite.
-				// Still need to ensure that preparer launches correctly
-				// as root
-				if pod.Id == POD_ID {
-					pod.RunAs = "root"
-				}
-
-				ok := p.installAndLaunchPod(&manifestToLaunch, pod, manifestLogger)
-				if ok {
-					manifestToLaunch = pods.Manifest{}
+				if (manifestToLaunch.RunAsUser() == "root") && (pod.Id != POD_ID) {
```

Comment:
I was kind of wondering if this logic should go inside of installAndLaunchPod, but I guess if you call that it should mean you have the intention of launching it. Also the way we've redefined the Pod interface in this file means I can't get to pod.RunAs or pod.Id in the other function